src/append/rolling_file/mod.rs

```diff
@@ -272,11 +295,86 @@ impl RollingFileAppenderBuilder {
             fs::create_dir_all(parent)?;
         }
 
-        // open the log file immediately
+        // Open the log file immediately
         appender.get_writer(&mut appender.writer.lock())?;
 
         Ok(appender)
     }
+
+    /// Constructs a `RollingFileAppender`.
+    /// The path argument can contain environment variables of the form $ENV{name_here},
+    /// where 'name_here' will be the name of the environment variable that
+    /// will be resolved. Note that if the variable fails to resolve,
+    /// $ENV{name_here} will NOT be replaced in the path. If set in the configuration, this
+    /// will roll the log file on startup as well.
+    #[cfg(feature = "compound_policy")]
+    pub fn build<P>(
```

Comment:
I think this function should be the same with or without compound policy. It's just the `if self.roll_on_startup` that should be conditionally compiled right?

Reply:
I think actually that block should just be a method like try_startup_roll which will roll if `self.roll_on_startup`

Reply:
Good point. I fixed it as good as I could. See my comment below for more details.